src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +59,123 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IPropertySetterConfiguration<TValue> CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+
+            return new PropertySetterConfiguration<TValue>(setterExpression, this.CreateVoidArgumentValidationConfiguration);
+        }
+
+        private static string GetIndexerName(MethodCallExpression methodCallExpression)
+        {
+            if (methodCallExpression != null &&
+                HasThis(methodCallExpression) &&
+                methodCallExpression.Method.IsSpecialName)
+            {
+                var methodName = methodCallExpression.Method.Name;
+                if (methodName.StartsWith("get_", StringComparison.Ordinal))
+                {
+                    return methodName.Substring(4);
+                }
+            }
+
+            return null;
+        }
+
+        private static bool HasThis(MethodCallExpression methodCallExpression)
+        {
+            return (methodCallExpression.Method.CallingConvention & CallingConventions.HasThis) ==
+                   CallingConventions.HasThis;
+        }
+
+        private static MethodCallExpression BuildSetterFromMemberExpression<TValue>(MemberExpression memberExpression)
+        {
+            PropertyInfo propertyInfo = memberExpression.Member as PropertyInfo;
+            if (propertyInfo == null)
+            {
+                throw new ArgumentException("The expression refers to '" + memberExpression.Member.Name +
+                                            "', which is a field, not a property getter.");
+            }
+
+            var propertySetterInfo = propertyInfo.GetSetMethod(nonPublic: true);
+            if (propertySetterInfo == null)
+            {
+                throw new ArgumentException("The property '" + memberExpression.Member.Name + "' does not have a setter.");
+            }
+
+            var arguments = new[] { BuildArgumentThatMatchesAnything<TValue>() };

```

Comment:
All the `Expression.Property`s I see take 2 arguments. I don't doubt you, but I'm confused.
Do you prefer the new version (if so, teach me why!), or was that just for education?